src/hugr.rs

```diff
@@ -149,6 +152,9 @@ impl Hugr {
     }
 }
 
+/// Arbitrary metadata for an operation.
+pub type OpMetadata = serde_json::Value;
```

Comment:
This is really metadata for a _node_, which may or may not be an _operation_. I suggest calling it `NodeMetadata` instead?